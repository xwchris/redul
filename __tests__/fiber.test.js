import Redul from '../src/index'
const { render} = Redul

const getRootNode = () => {
    document.body.innerHTML = '<div id="root"></div>'
    const $root = document.getElementById('root')
    return $root
}
const $root = getRootNode()

describe('fiber reconcile test', () => {
    describe('mount stage', () => {
        test('element string type', () => {

            render(<div className="test" style={{color: 'red'}}></div>, $root)
            expect($root).toMatchSnapshot()
        })

        test('element function type', () => {
            function Count({ count }) {
                return <div>{count}</div>
            }

            render(<Count count={1} />, $root)
            expect($root).toMatchSnapshot()
        })

        test('element complex function type', () => {
            function App() {
                return <div><Count count={1} /></div>
            }

            function Count({ count }) {
                return <span>{count}</span>
            }

            render(<App />, $root)
            expect($root).toMatchSnapshot()
        })
    })
})
