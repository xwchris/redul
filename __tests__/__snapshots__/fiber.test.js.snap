// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fiber reconcile test mount stage element complex function type 1`] = `
<div
  id="root"
>
  <div>
    <span>
      1
    </span>
  </div>
</div>
`;

exports[`fiber reconcile test mount stage element function type 1`] = `
<div
  id="root"
>
  <div>
    1
  </div>
</div>
`;

exports[`fiber reconcile test mount stage element string type 1`] = `
<div
  id="root"
>
  <div
    class="test"
    style="color: red"
  />
</div>
`;
